mlfinlab/online_portfolio_selection/online_portfolio_selection.py

```diff
@@ -0,0 +1,296 @@
+# pylint: disable=missing-module-docstring
+import pandas as pd
+import numpy as np
+import cvxpy as cp
+
+
+class OLPS:
+    """
+    Online Portfolio Selection is an algorithmic trading strategy that sequentially allocates capital among a group of
+    assets to maximize the final returns of the investment.
+
+    Traditional theories for portfolio selection, such as Markowitz’s Modern Portfolio Theory, optimize the balance
+    between the portfolio's risks and returns. However, OLPS is founded on the capital growth theory, which solely
+    focuses on maximizing the returns of the current portfolio.
+
+    Through these walkthroughs of different portfolio selection strategies, we hope to introduce a set of different
+    selection tools available for everyone. Most of the works will be based on Dr. Bin Li and Dr. Steven Hoi’s book,
+    Online Portfolio Selection: Principles and Algorithms, and further recent papers will be implemented to assist the
+    development and understanding of these unique portfolio selection strategies.
+
+    OLPS is the parent class for all resulting Online Portfolio Selection Strategies.
+
+    This class broadly defines all variables and allocates a set of weights given a certain strategy
+
+    :ivar weights: (np.array) final portfolio weights prediction
+    :ivar all_weights: (pd.DataFrame) portfolio weights for the time period
+    :ivar asset_name: (list) name of assets
+    :ivar number_of_assets: (int) number of assets
+    :ivar time: (datetime) time index of the given data
+    :ivar length_of_time: (int) number of time periods
+    :ivar relative_return: (np.array) relative returns of the assets
+    :ivar portfolio_return: (pd.DataFrame) cumulative portfolio returns over time
+    :ivar asset_prices: (pd.DataFrame) a dataframe of historical asset prices (daily close)
+    """
+
+    def __init__(self):
+        self.weights = None
+        self.all_weights = None
+        self.asset_name = None
+        self.number_of_assets = None
+        self.time = None
+        self.length_of_time = None
+        self.relative_return = None
+        self.portfolio_return = None
+        self.asset_prices = None
+
+    def allocate(self,
```

Comment:
I prefer these to be on a single line, unless there is a PEP standard I am not aware of?